File: algorithms/implementation/angry-professor/angry-professor-javascript/js/test_angryProfessor.js
import {angryProfessor} from '../js/angryProfessor.js';
import assert from 'assert';

describe('angryProfessor', function() {

    it('A aula será cancelada', function() {

        let k = 3;
        let a = [-1, -3, 4, 2];

        let esperado = "YES";

        let resultado = angryProfessor(k, a);

        assert.strictEqual(resultado, esperado);
        
    });
});

describe('angryProfessor', function() {

    it('A aula não será cancelada', function() {

        let k = 2;
        let a = [0, -1, 2, 1];

        let esperado = "NO";

        let resultado = angryProfessor(k, a);

        assert.strictEqual(resultado, esperado);
        
    });
});


describe('angryProfessor', function() {

    it('A aula não será cancelada', function() {

        let k = 4;
        let a = [0, -1, 2, -2, 1, -3];

        let esperado = "NO";

        let resultado = angryProfessor(k, a);

        assert.strictEqual(resultado, esperado);
        
    });
});

File: algorithms/implementation/angry-professor/angry-professor-python/tests/test_angryProfessor.py
from angryProfessor import angryProfessor

class TesteAngryProfessor:
    
    def testeCaso1(self):
        
        k = 3
        a = [-1, -3, 4, 2]
        
        esperado = "YES"
        
        resultado = angryProfessor(k, a)
        
        assert resultado == esperado
        
    def testeCaso2(self):
        
        k = 2
        a = [0, -1, 2, 1]
        
        esperado = "NO"
        
        resultado = angryProfessor(k, a)
        
        assert resultado == esperado